Doc/reference/datamodel.rst

```diff
@@ -1341,7 +1341,12 @@ Special read-only attributes
 
    * - .. attribute:: frame.f_locals
      - The dictionary used by the frame to look up
-       :ref:`local variables <naming>`
+       :ref:`local variables <naming>`.
+       If the frame refers to a function or comprehension,
+       this may return a write-through proxy object.
+
+       .. versionchanged:: 3.13
+          Return a proxy for functions and comprehensions.
```

Comment:
And apparently this also applies for type parameter scopes? Though I have no idea if that was deliberate:

```pycon
>>> import sys
>>> type T[X: type(sys._getframe().f_locals)] = ...
>>> T.__type_params__[0].__bound__
<class 'FrameLocalsProxy'>
```